### PR TITLE
feat(ui/console): add copy as curl button

### PR DESCRIPTION
**Pull request by AmineRhazzar**
fixes: #2255 

- implemented 'Copy as curl' button
- Fixed the following ui bugs:
    + Request Context editor hides the 'Evaluate' button on smaller heights
    + The component showing the response gets highlighted (using hljs) only on the first request, and not on subsequent requests or when an error happens.

I wasn't sure how to specify the need for an API token for the curl command if auth was required. I just put `-H "Authorization: Bearer <api-token>"`.

### CODE REVIEW

**Comment by erka** [comment]
@AmineRhazzar could you please change `Refs: #2255` to `fixes: #2255`. Github [will link issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) with current PR and close the issue when PR is merged.

**Comment by GeorgeMac** [comment]
This is really awesome, thank you @AmineRhazzar 🙏 

I think what you have there for auth makes sense. We could have a little input field for supplying a token, but unsure if we really want that and could always come in a later change if we decided we want it.

**Review by GeorgeMac — APPROVED** [review]
This looks awesome and thanks for fixing the highlighting bug too!

**Comment by AmineRhazzar** [comment]
You're welcome! Happy to help

File: ui/src/types/Auth.ts [hunk 1]
```diff
@@ -1,5 +1,9 @@
 export interface IAuthMethod {
-  method: 'METHOD_TOKEN' | 'METHOD_OIDC' | 'METHOD_GITHUB';
+  method:
+    | 'METHOD_TOKEN'
+    | 'METHOD_OIDC'
+    | 'METHOD_GITHUB'
+    | 'METHOD_KUBERNETES';
```
Comment (markphelps) [inline ui/src/types/Auth.ts:6]: good catch!

File: ui/src/app/console/Console.tsx [hunk 2]
```diff
@@ -110,9 +141,54 @@ export default function Console() {
...
+        namespaceKey: namespace.key
+      },
+      headers,
+      uri
+    });
+
+    copyTextToClipboard(command);
+
+    setSuccess(
+      'Command copied to clipboard. Generate an API token if necessary.'
```
Comment (markphelps) [inline ui/src/app/console/Console.tsx:180]:
```suggestion
      'Command copied to clipboard'
```

Comment (markphelps) [reply ui/src/app/console/Console.tsx:180]: I think the 'Generate API Token' bit is self explanatory

File: ui/src/app/console/Console.tsx [hunk 3]
```diff
@@ -223,6 +303,17 @@ export default function Console() {
                         </div>
                       </div>
                       <div className="flex justify-end">
+                        <Button
+                          primary
```
Comment (markphelps) [inline ui/src/app/console/Console.tsx:307]:
```suggestion
```
I dont think we want two primary buttons, the 'Evaluate' should still be the primary IMO and this secondary

**Review by markphelps — COMMENTED** [review]
couple minor suggestions @AmineRhazzar 

thank you very much for doing this! works great!

**Comment by markphelps** [comment]
@all-contributors please add @AmineRhazzar for code

**Review by markphelps — APPROVED** [review]
thanks again @AmineRhazzar !!

**Comment by AmineRhazzar** [comment]
My pleasure 👌

**Pull request merged**